aqueduct/gateway/views/decorators.py

```diff
@@ -29,6 +32,92 @@
 logger = logging.getLogger("aqueduct")
 
 
+def mcp_transport_security(view_func):
+    """Validate MCP transport security (DNS rebinding protection).
+
+    Validates:
+    - Host header (DNS rebinding protection)
+    - Origin header (CSRF protection)
+    - Content-Type header for POST requests
+
+    Returns appropriate status codes:
+    - 421: Invalid Host header
+    - 403: Invalid Origin header
+    - 400: Invalid Content-Type header
+    """
+
+    @wraps(view_func)
+    async def wrapper(request: ASGIRequest, *args, **kwargs):
```

Comment:
I would consider splitting this function into smaller ones, each one validating one thing, and with a docstring: one with the host-matching logic, one with the headers validation, etc.